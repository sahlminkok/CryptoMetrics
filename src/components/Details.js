import React from 'react';
import { useParams, Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import backArrow from '../assets/arrowLeft.svg';
import style from '../styles/Details.module.css';

const Details = () => {
  const { id } = useParams();
  const { filteredCryptos } = useSelector((store) => store.cryptos);
  const crypto = filteredCryptos.find((crypto) => crypto.id === id);

  if (!crypto) {
    return <h2>Coin not found</h2>;
  }

  return (
    <div>
      <div className={style.title}>
        <div className={style.backArrow}>
          <Link to="/">
            <img src={backArrow} alt="back" />
          </Link>
        </div>
        <div className={style.name}>
          <h1>{crypto.name}</h1>
          <p>{crypto.symbol}</p>
        </div>
      </div>
      <div className={style.details}>
        <p className={style.card}>
          <span>Rank:</span>
          <span>{Number(crypto.rank)}</span>
        </p>
        <p className={style.card}>
          <span>Supply:</span>
          <span>{Number(crypto.supply)}</span>
        </p>
        <p className={style.card}>
          <span>Max Supply:</span>
          <span>{Number(crypto.maxSupply)}</span>
        </p>
        <p className={style.card}>
          <span>Market Cap:</span>
          <span>
            $
            {Number(crypto.marketCapUsd).toFixed(2)}
          </span>
        </p>
        <p className={style.card}>
          <span>Volume 24h:</span>
          <span>{Number(crypto.volumeUsd24Hr).toFixed(2)}</span>
        </p>
        <p className={style.card}>
          <span>Price:</span>
          <span>
            $
            {Number(crypto.priceUsd).toFixed(6)}
          </span>
        </p>
        <p className={style.card}>
          <span>Change Percentage:</span>
          <span>
            {Number(crypto.changePercent24Hr).toFixed(2)}
            %
          </span>
        </p>
      </div>
    </div>
  );
};

export default Details;
