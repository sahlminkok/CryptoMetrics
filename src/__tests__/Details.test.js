import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Details from '../components/Details';

// Create a mock Redux store
const mockStore = configureStore([]);
const store = mockStore({
  cryptos: {
    filteredCryptos: [
      {
        id: '1', // Assuming this matches the ID parameter used in the test
        name: 'Bitcoin',
        symbol: 'BTC',
        rank: 1,
        supply: 21000000,
        maxSupply: 21000000,
        marketCapUsd: 123456789,
        volumeUsd24Hr: 987654321,
        priceUsd: 40000,
        changePercent24Hr: 1.5,
      },
    ],
  },
});

test('renders Details component correctly', () => {
  // Render the component with the mock store and router
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/details/1']}>
        <Routes>
          <Route path="/details/:id" element={<Details />} />
        </Routes>
      </MemoryRouter>
    </Provider>,
  );

  // Assert that the rendered component matches the snapshot
  expect(container).toMatchSnapshot();
});
