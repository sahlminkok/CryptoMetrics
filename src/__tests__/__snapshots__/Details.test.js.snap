// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Details component correctly 1`] = `
<div>
  <div
    data-testid="details"
  >
    <div
      class="title"
    >
      <div
        class="backArrow"
      >
        <a
          href="/"
        >
          <img
            alt="back"
            src="arrowLeft.svg"
          />
        </a>
      </div>
      <div
        class="name"
      >
        <h1>
          Bitcoin
        </h1>
        <p>
          BTC
        </p>
      </div>
    </div>
    <div
      class="details"
    >
      <p
        class="card"
      >
        <span>
          Rank:
        </span>
        <span>
          1
        </span>
      </p>
      <p
        class="card"
      >
        <span>
          Supply:
        </span>
        <span>
          21000000
        </span>
      </p>
      <p
        class="card"
      >
        <span>
          Max Supply:
        </span>
        <span>
          21000000
        </span>
      </p>
      <p
        class="card"
      >
        <span>
          Market Cap:
        </span>
        <span>
          $
          123456789.00
        </span>
      </p>
      <p
        class="card"
      >
        <span>
          Volume 24h:
        </span>
        <span>
          987654321.00
        </span>
      </p>
      <p
        class="card"
      >
        <span>
          Price:
        </span>
        <span>
          $
          40000.000000
        </span>
      </p>
      <p
        class="card"
      >
        <span>
          Change Percentage:
        </span>
        <span>
          1.50
          %
        </span>
      </p>
    </div>
  </div>
</div>
`;
